erc20-price/src/lib.rs

```diff
@@ -188,3 +190,157 @@ fn store_chainlink_price(
+            };
+
+            output.set(
+                0,
+                keyer::pair_info_key(&pair_created_event.pair),
+                &pair_created_event,
+            );
+        }
+    }
+}
+
+#[substreams::handlers::store]
+fn store_uniswap_price(
+    block: eth::Block,
+    chainlink_prices: StoreGetProto<Erc20Price>,
+    store: StoreGetProto<PairCreatedEvent>,
+    output: StoreSetProto<Erc20Price>,
+) {
+    for log in block.logs() {
+        if let Some(event) = pair::events::Sync::match_and_decode(log) {
+            let pair_address = Hex(log.address()).to_string();
+
+            if let Some(pair) = store.get_last(keyer::pair_info_key(&pair_address)) {
+                let reserve0 = event
+                    .reserve0
+                    .to_decimal(pair.token0.clone().unwrap().decimals);
+                let reserve1 = event
+                    .reserve1
+                    .to_decimal(pair.token1.clone().unwrap().decimals);
+
+                // TODO: Add a check for mininmum liquidity threshhold.
+
+                match pair.token0.clone().unwrap().symbol.as_str() {
+                    symbol if ["DAI", "USDC", "USDT"].contains(&symbol) => {
+                        let token_price = reserve0.clone() / reserve1.clone();
+
+                        let erc20price = Erc20Price {
+                            token: pair.token1.clone(),
+                            price_usd: format!("{:.7}", token_price.to_string()),
+                            block_number: block.number,
+                            source: 2,
+                        };
+
+                        output.set(
+                            0,
+                            keyer::uniswap_asset_key(&pair.token0.clone().unwrap().address),
+                            &erc20price,
+                        );
+                    }
+                    "WETH" => {
+                        let eth_price = match chainlink_prices.get_last(keyer::chainlink_asset_key(
+                            &pair.token0.clone().unwrap().address,
+                        )) {
+                            Some(price) => BigDecimal::from_str(price.price_usd.as_str()).unwrap(),
+                            None => BigDecimal::zero(),
+                        };
+
+                        if eth_price.eq(&BigDecimal::zero()) {
+                            break;
+                        };
+
+                        let token_price = (reserve0.clone() / reserve1.clone()) * eth_price;
+
+                        let erc20price = Erc20Price {
+                            token: pair.token1.clone(),
+                            price_usd: format!("{:.7}", token_price.to_string()),
+                            block_number: block.number,
+                            source: 2,
+                        };
+
+                        output.set(
+                            0,
+                            keyer::uniswap_asset_key(&pair.token0.clone().unwrap().address),
```

Comment:
i don't think you can assume the key to be always `token0`, you will need a check here

Reply:
yeah, that's a typo. The last output set should be using `token1`.